tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.ScenarioTests.ps1

```diff
@@ -0,0 +1,345 @@
+        $profilesInstalled = Invoke-Command -Session $session -ScriptBlock { Get-AzureRmProfile } 
+        $profilesInstalled.Contains('Latest') | Should Be $true
+
+        # Act
+        Invoke-Command -Session $session -ScriptBlock { Use-AzureRmProfile -Profile 'Latest' }
+        $result = Invoke-Command -Session $session -ScriptBlock { Get-Module -FullyQualifiedName 'AzureRm' }
+
+        # Get the version of the Latest profile
+        $ProfileMap = Get-AzProfile
+        $latestVersion = $ProfileMap.'Latest'.$RollupModule
+
+        # Assert
+        It "Should return AzureRm module Latest version" {
+            $result.Name | Should Be $RollupModule
+            $result.version | Should Be $latestVersion
+        }
+
+        # Cleanup
+        Invoke-Command -Session $session -ScriptBlock { Uninstall-AzureRmProfile -Profile 'Latest' -Force -ea SilentlyContinue }
+        Remove-PSSession -Session $session
+    }
+}
+
+Describe "User can update their machine to a latest profile" {
+    Context "Profile 2016-09 is installed" {
+        # Should refresh profile map from Azure end point and update modules.
+        # Arrange
+        # Create a new PS session
+        $session = New-PSSession
+
+        # Check if '2016-09' is installed
+        $profilesInstalled = Invoke-Command -Session $session -ScriptBlock { Get-AzureRmProfile } 
+        $profilesInstalled | Should Be '2016-09'
+
+        # Act
+        Invoke-Command -Session $session -ScriptBlock { Get-AzureRmProfile -Update }
+        Invoke-Command -Session $session -ScriptBlock { Use-AzureRmProfile -Profile 'Latest' -Force }
+        $result = Invoke-Command -Session $session -ScriptBlock { Get-AzureRmProfile }
+        $modules = Invoke-Command -Session $session -ScriptBlock { Get-Module -FullyQualifiedName 'AzureRm' }
+
+        # Assert
+        It "Should return 2016-09 & Latest" {
+            $result.Length | Should Be 2
+            $result.Contains('2016-09') | Should Be $true 
+            $result.Contains('Latest') | Should Be $true
+        }
+
+        It "Latest version of modules are imported" {
+            # Get the version of the Latest profile
+            $ProfileMap = Get-AzProfile
+            $latestVersion = $ProfileMap.'Latest'.$RollupModule
+            
+            # Are latest modules imported?
+            $modules.Name | Should Be $RollupModule
+            $modules.version | Should Be $latestVersion
+        }
+
+        It "Last Write Time should be less than 5 minutes" {
+            # Get LastWriteTime for ProfileMap
+            $lastWriteTime = (Get-Item -Path (Join-Path $Env:LocalAppData -ChildPath 'Microsoft\AzurePowerShell\ProfileCache\ProfileMap.json')).LastWriteTime
+            (((Get-Date) - $lastWriteTime).TotalMinutes -lt 5) | Should Be $true
+        }
+
+        # Cleanup
+        Remove-PSSession -Session $session
+    }
+}
+
+Describe "User can uninstall a profile" {
+    Context "Latest profile is installed" {
```

Comment:
TODO need scenarios for: (1) removing a named profile when multiple profiles are installed; (2) Uninstalling all profiles